service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -134,14 +141,24 @@ protected void setCollaborators(
             adapter.setCommandConsumerFactory(commandConsumerFactory(adapterProperties, samplerFactory, commandRouterClient));
         }
 
+        final KafkaProducerConfigProperties kafkaProducerConfig = kafkaProducerConfig();
```

Comment:
does this work (i.e. doesn't throw a runtime exception) if no `hono.kafka` properties are set at all?

Reply:
Yes. That is the case when `KafkaProducerConfigProperties.getProducerConfig()` returns `null` (see discussion above).

But I would prefer to use the change in #2326 which would make this construct obsolete.